draft-lenders-core-coap-dtls-svcb.md

```diff
@@ -60,61 +61,52 @@ normative:
 informative:
   RFC8323: coap-tcp
   I-D.ietf-core-dns-over-coap: doc
+  RFC4944: 6lo
 
 --- abstract
 
-This document specifies the usage of Service Parameters as used in SVCB ("Service Binding") DNS resource
-records for the discovery of transport-layer-secured CoAP services.
+This document specifies an Application-Layer Protocol Negotiation (ALPN) ID for
+transport-layer-secured CoAP services.
 
 --- middle
 
 # Introduction
 
-{{-svcb}} specifies the "SVCB" ("Service Binding") DNS resource records for looking up
-communication endpoints of a service. Service Parameters (SvcParams) are used to
-carry that information.
-This document specifies which information from SvcParams can be used with CoAP services that are
-secured by transport security, namely TLS and DTLS.
-As an example, this information can be obtained as part of the discovery of DNS over CoAP (DoC) servers (see
-{{-doc}}) that deploy TLS or DTLS to secure their messages.
+Application-Layer Protocol Negotiation (ALPN) allows agreeing on an application-layer protocol during a Transport Layer Security (TLS) handshake.
+For that an ALPN ID is used.
+Furthermore, the ALPN ID can be used to discover Service Bindings (SVCB) via DNS, using SVCB resource records with the "alpn" Service Parameter Keys.
+As an example, this information can be obtained as part of the discovery of DNS over CoAP (DoC) servers (see {{-doc}}) that deploy TLS or DTLS to secure their messages.
+This document specifies ALPN IDs for CoAP services that are secured by transport security, namely TLS and DTLS.
 
 # Terminology
 
-SvcParams denotes the field in either DNS SVCB/HTTPS records as defined in {{-svcb}}, or DHCP and RA
-messages as defined in {{-dnr}}.
-
 {::boilerplate bcp14-tagged}
 
 # Application-Layer Protocol Negotiation (ALPN) IDs
 
-{{-svcb}} defines the "alpn" key, which is used to identify the service binding to a protocol suite
-using its Application-Layer Protocol Negotiation (ALPN) ID {{-alpn}}.
 For CoAP over TLS an ALPN ID was defined in {{-coap-tcp}}.
 As it is not advisable to re-use the same ALPN ID for a different transport layer, an ALPN for
 CoAP over DTLS is also registered in {{iana}}.
+
+ALPN ID values have no fixed length.
+A short value ("co") is allocated for CoAP over DTLS.
+This avoids fragmentation of Client Hello and Server Hello messages in constrained networks with link-layer fragmentation, such as 6LoWPAN {{-6lo}}.
```

Comment:
```suggestion
This can avoid fragmentation of Client Hello and Server Hello messages in constrained networks with link-layer fragmentation, such as 6LoWPAN {{-6lo}}.
```

We don't have concrete evidence of this being possible right now. (If our last experiments were optimized correctly, an ALPN wouldn't have made a difference; if cTLS hits, that should be different).